multirowgroup.go

```diff
@@ -0,0 +1,246 @@
+package parquet
+
+import (
+	"io"
+)
+
+// MultiRowGroup wraps multiple row groups to appear as if it was a single
+// RowGroup. RowGroups must have the same schema or it will error.
+func MultiRowGroup(rowGroups []RowGroup) (RowGroup, error) {
+	if len(rowGroups) == 0 {
+		return nil, nil
+	}
+	if len(rowGroups) == 1 {
+		return rowGroups[0], nil
+	}
+
+	c := new(multiRowGroup)
+	c.init(rowGroups)
+	return c, nil
+}
+
+func (c *multiRowGroup) init(rowGroups []RowGroup) error {
+	schema, err := compatibleSchema(rowGroups)
+	if err != nil {
+		return err
+	}
+
+	c.schema = schema
+	c.rowGroups = rowGroups
+	c.columns = make([]multiColumnChunk, numLeafColumnsOf(schema))
+
+	for i := range c.columns {
+		c.columns[i].rowGroup = c
+		c.columns[i].column = i
+		c.columns[i].chunks = make([]ColumnChunk, len(rowGroups))
+
+		for j, rowGroup := range rowGroups {
+			c.columns[i].chunks[j] = rowGroup.Column(i)
+		}
+	}
+
+	return nil
+}
+
+func compatibleSchema(rowGroups []RowGroup) (*Schema, error) {
```

Comment:
To follow the naming scheme of functions with similar purpose:
```suggestion
func compatibleSchemaOf(rowGroups []RowGroup) (*Schema, error) {
```